### ISSUE TITLE
Version mismatch on latest release (2.28.5)

### ISSUE THREAD

**Issue by cclark25** [libsdl-org/SDL#8538]
I am having trouble with the SDL versions not matching between the headers and compiled library, and I would greatly appreciate any help.

I have downloaded the 2.28.5 release package, unzipped it, and ran `./configure`, `./make`, and `DESTDIR=/home/USER/Downloads/SDL2-2.28.5/testinstall ./make install` to install the SDL library locally.  
I then manually copied the generated **bin**, **include**, **lib**, and **share** folders into my project at `dependencies/cpp/sdl/linux` to keep a local copy of the dependency in my project.

Within my C++ app, I am getting an error from `SDL_GetWindowWMInfo` saying "Application not compiled with SDL 2.0". After debugging this I found that the `SDL_VERSION` macro is producing the expected version number 2.28.5, but the `SDL_GetVersion` function is producing the version number 2.0.10. So it appears that I am somehow linking the wrong version of SDL, even though I am building it from the release package.

My compilation command produced by my Makefile is:
>g++ -fpic -shared \
        build/cpp-headers/game-engine.cxx \
        src/cpp/test.cpp src/cpp/error.cpp src/cpp/platform-specific/window/window.cpp \
        -Ldependencies/cpp/sdl/linux/lib \
        -I./src/swig-headers \
        -Idependencies/cpp/sdl/linux/include/SDL2 \
        -o build/output/GameEngine \
        -Wl,--enable-new-dtags -lSDL2

I have experimented with using the `sdl2-config` script to generated all the needed command line arguments, without much luck. I am pretty sure that this compiler command is using the ".so" file in `dependencies/cpp/sdl/linux/lib`, because SDL is not installed globally on my system and removing the `-Ldependencies/cpp/sdl/linux/lib` line breaks linkage.

So my conclusion is that for whatever reason, the 2.28.5 is building the library with the 2.0.10 version number while the header files are using the valid 2.28.5 version number.

For some extra context:
I am developing a game engine using SDL to create a window. That part works.
I am trying to use the `SDL_GetWindowWMInfo` function in order to pull out the native window handle, which I will pass into the [BGFX library](https://github.com/bkaradzic/bgfx) to perform the drawing operations on. I have not linked in the BGFX library yet, so it can't be related.
The C++ code is all compiled into a shared library with binding information generated by SWIG to expose behavior to a C# runtime. This is included in the compilation command with `build/cpp-headers/game-engine.cxx`.
This is on Linux Mint 20.1 Cinnamon, the kernel is 5.4.0-155-generic.

**Comment by slouken** [comment]
It sounds like you're picking up the version of SDL installed on the system.

**Comment by madebr** [comment]
You may not have installed the SDL2 development libraries, provided by your distro, it looks like you have installed SDL2 runtime libraries (`/usr/lib64/libSDL2-2.0.so.0`).

Try adding the lib folder of your custom built SDL to your runpath:
```
-Wl,-rpath,/full/path/to/dependencies/cpp/sdl/linux/lib
```

**Comment by cclark25** [comment]
Okay I managed to get my application working, after @madebr 's suggestion put me on the right track.
I had already removed all global instances of libSDL, so that was not the issue.

After adding @madebr 's suggestion of` -Wl,-rpath,/full/path/to/dependencies/cpp/sdl/linux/lib`, the SDL error was gone and instead I was getting a segmentation fault, which was caused by an unrelated piece of code after the `SDL_GetWindowWMInfo` call.

Before I realized the cause of the segmentation fault, I was assuming that the `SDL_GetWindowWMInfo` call was causing it. I saw this fault before while experimenting with the command line arguments, but mistook it for an error with linking SDL.

Now that I have removed the faulty code, the app is running as expected, with no SDL error or seg fault.

Thank you guys for your help!
